agent/stats/engine.go

```diff
@@ -123,7 +124,9 @@ func NewDockerStatsEngine(cfg *config.Config, client ecsengine.DockerClient, con
 
 // synchronizeState goes through all the containers on the instance to synchronize the state on agent start
 func (engine *DockerStatsEngine) synchronizeState() error {
-	listContainersResponse := engine.client.ListContainers(false, ecsengine.ListContainersTimeout)
+	ctx, cancel := context.WithTimeout(context.Background(), ecsengine.ListContainersTimeout)
```

Comment:
Here too, you shouldn't need to create a new context. Just use `engine.ctx` as the parent and let the child context creation happen in `ListContainers`

Reply:
There is no ctx attribute for stats engine, and I think creating a context for just one method seems to be redundant. So instead of creating a timeout context from background context, I will pass the background context directly to the method.

Reply:
You can create one :) Let it be set from app.ctx.

Reply:
ok, cool